sspl_test/alerts/self_hw/test_node_bmc.py

```diff
@@ -19,38 +19,46 @@
 
 import os
 import re
-import socket
 
 from framework.utils.conf_utils import (
     Conf, SSPL_CONF, GLOBAL_CONF, MACHINE_ID, BMC_IP_KEY,
     BMC_USER_KEY, BMC_SECRET_KEY)
-from alerts.self_hw.self_hw_utilities import get_manufacturer_name
+from alerts.self_hw.self_hw_utilities import get_server_details
 from cortx.utils.process import SimpleProcess
 from cortx.utils.security.cipher import Cipher
 from cortx.utils.validator.v_bmc import BmcV
+from cortx.utils.ssh import SSHChannel
 
 
 CHANNEL_PROTOCOL = "Channel Protocol Type"
 CHANNEL_MEDIUM = "Channel Medium Type"
 FIRMWARE_VERSION = "Firmware Revision"
+IPMI_VERSION = "IPMI Version"
+REQUIRED_IPMI_VERSION = "2.0"
+SEL_VERSION = "Version"
+MIN_REQUIRED_SEL_VERSION = 2
+
 
 def init(args):
     pass
 
 def test_bmc_config(args):
-    """Check if BMC configuration are valid."""
-    pkey_file = "/root/.ssh/id_rsa"
-    if not os.path.exists(pkey_file):
-        print("\tSkipping test_bmc_config.. pkey file not found.")
-        return
+    """Check if BMC configuration are valid.
+
+    Testing BMC config with ipmitool is possible only when ipmi over lan
+    is configured(out-band setup). It is taken care by test_bmc_is_accessible.
+    So, validation on bmc onfiguration with bmc ip, user and secret value
+    through ssh is fine at this time.
```

Comment:
sure. no ssh key file requirement now !?

Reply:
Yes, I avoided that.
This test_bmc_config() is now just validating given BMC IP, user, passwd.
So ssh key file dependency is not required.

test_bmc_is_accessible() method is validating rc of ipmitool with that given config.
```cmd = "sudo ipmitool -H %s -U %s -P %s -I lan %s" %(
            bmc_ip, bmc_user, bmc_passwd, subcommand)
```

Reply:
Done.

Reply:
good